### PR TITLE
Mon 3568 host filters

### PR DESCRIPTION
**Pull request by miteto**
<h1> Host filters </h1>

<h2> Fix host filters </h2>

Fixes 3568

<h2> Type of change </h2>

- [X] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

<h2> Target serie </h2>

- [ ] 2.8.x
- [X] 18.10.x
- [X] 19.04.x (master)

### CODE REVIEW

File: www/include/configuration/configObject/host/listHost.php [hunk 1]
```diff
@@ -82,7 +82,7 @@
     FILTER_VALIDATE_INT
 );
 
-if (isset($_POST['searchH']) || $_GET['searchB']) {
+if (isset($_POST['searchH']) || isset($_GET['searchB'])) {
     //saving chosen filters values
     $num = 0;
```
Comment (sc979) [inline www/include/configuration/configObject/host/listHost.php]:
```suggestion
```

File: www/include/configuration/configObject/host/listHost.php [hunk 2]
```diff
@@ -94,9 +94,9 @@
 } else {
     //restoring saved values
     $search = $centreon->historySearch[$url]['searchH'] ?? null;
-    $poller = (int)$centreon->historySearch[$url]["poller"] ?? 0;
-    $hostgroup = (int)$centreon->historySearch[$url]["hostgroup"] ?? 0;
-    $template = (int)$centreon->historySearch[$url]["template"] ?? 0;
+    $poller = (int)($centreon->historySearch[$url]["poller"] ?? 0);
```
Comment (sc979) [inline www/include/configuration/configObject/host/listHost.php]: as "0" is an integer, is it usefull to add the parenthesis ?

File: www/include/configuration/configObject/host/listHost.php [hunk 3]
```diff
@@ -94,9 +94,9 @@
...
+    $hostgroup = (int)($centreon->historySearch[$url]["hostgroup"] ?? 0);
+    $template = (int)($centreon->historySearch[$url]["template"] ?? 0);
 }
```
Comment (sc979) [inline www/include/configuration/configObject/host/listHost.php]: please can you add  : 
`$status = $centreon->historySearch[$url]["status"];` I removed it by mistake, when I was trying to debug the code to make the acceptance test :'(

File: www/include/configuration/configObject/host/listHost.ihtml [hunk 4]
```diff
@@ -68,7 +68,7 @@
 			</a>
 		</td>
 		<td class="ListColCenter">
-			<a href='./main.php?p=602&search_h={$elemArr[elem].RowMenu_name}&search_s='>
+			<a href='./main.php?p=602&searchH={$elemArr[elem].RowMenu_name}&Search='>
```
Comment (sc979) [inline www/include/configuration/configObject/host/listHost.ihtml]:
```suggestion
			<a href='./main.php?p=602&searchH={$elemArr[elem].RowMenu_name}'>
```

Comment (sc979) [reply www/include/configuration/configObject/host/listHost.ihtml]: as $Search is always empty, is it useful to keep it ?

File: www/include/configuration/configObject/host/listHost.php (hunk 2 above)
Comment (miteto) [reply www/include/configuration/configObject/host/listHost.php]: This fixes PHP warnings about  non-existing index.

File: www/include/configuration/configObject/host/listHost.php (hunk 3 above)
Comment (miteto) [reply www/include/configuration/configObject/host/listHost.php]: Done

File: www/include/configuration/configObject/host/listHost.php [hunk 5]
```diff
@@ -94,9 +94,10 @@
 } else {
     //restoring saved values
     $search = $centreon->historySearch[$url]['searchH'] ?? null;
-    $poller = (int)$centreon->historySearch[$url]["poller"] ?? 0;
-    $hostgroup = (int)$centreon->historySearch[$url]["hostgroup"] ?? 0;
-    $template = (int)$centreon->historySearch[$url]["template"] ?? 0;
+    $poller = (int)($centreon->historySearch[$url]["poller"] ?? 0);
+    $hostgroup = (int)($centreon->historySearch[$url]["hostgroup"] ?? 0);
+    $template = (int)($centreon->historySearch[$url]["template"] ?? 0);
+    $status = (int)($centreon->historySearch[$url]["status"]);
```
Comment (sc979) [inline www/include/configuration/configObject/host/listHost.php]:
```suggestion
    $status = (int)$centreon->historySearch[$url]["status"] ?? -1;
```

**Pull request closed without merging**